src/main/java/com/giovds/AverageAgeMojo.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Mojo(name = "average",
+        defaultPhase = LifecyclePhase.VERIFY,
+        requiresOnline = true,
+        requiresProject = true,
+        requiresDependencyResolution = ResolutionScope.COMPILE)
+public class AverageAgeMojo extends AbstractMojo {
+    private QueryClient client = new QueryClient();
+
+    @Parameter(readonly = true, required = true, defaultValue = "${project}")
+    private MavenProject project;
+
+    private final LocalDate today = LocalDate.now();
+
+    // Necessary for Maven
+    public AverageAgeMojo() {
+    }
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        var isAggregatorProject = "pom".equals(project.getPackaging());
+        if (isAggregatorProject) {
+            getLog().info("Detected aggregator project");
+        }
+
+        final Set<Dependency> dependenciesToConsider = project.getArtifacts().stream()
+                .peek(artifact -> debug("Found artifact %s", artifact.getId()))
+                .filter(this::isCompileTimeDependency)
+                .peek(artifact -> debug("Found compile-time artifact %s", artifact.getId()))
+                .filter(artifact -> !isAggregatorProject || isDependencyOutsideProject(artifact))
+                .peek(artifact -> info("Found compile-time artifact outside this project %s", artifact.getId()))
+                .map(this::convertToDependency)
+                .collect(Collectors.toSet());
+
+        if (dependenciesToConsider.isEmpty()) {
+            info("No dependencies found");
+            return;
+        }
+
+        final List<String> queries = DependenciesToQueryMapper.mapToQueries(dependenciesToConsider);
+        debug("Queries: %s", queries);
+
+        final Set<DependencyResponse> results = client.search(queries);
+        debug("Results: %s", results);
+
+        final var stats = results.stream()
+                .sorted(Comparator.comparing(DependencyResponse::id))
+                .mapToLong(dependency -> {
+                    final long ageInDays = this.calculateDependencyAge(dependency);
+                    info("Dependency %s is %d days old", dependency.id(), ageInDays);
+                    return ageInDays;
+                })
+                .summaryStatistics();
+
+        var numberOfDependencies = dependenciesToConsider.size();
+        info("Found %d dependencies", numberOfDependencies);
```

Comment:
You could consider using `stats.count` instead of getting the size to more accurately represent the summary result, though I'd expect this to always be the same (how would have Maven resolved it otherwise). Either is fine for me, I'll leave this decision up to you.

Reply:
Makes sense.